src/main/java/org/craftercms/studio/impl/v2/repository/GitContentRepository.java

```diff
@@ -1354,6 +1354,57 @@ public boolean contentExists(String site, String path) {
         return toReturn;
     }
 
+    @Override
+    public String createFolder(String siteId, String path, String name) throws UserNotFoundException, ServiceLayerException {
+        Path emptyFilePath = Paths.get(path, name, EMPTY_FILE);
+        String newFolderPath = Paths.get(path, name).toString();
+        String gitLockKey = helper.getSandboxRepoLockKey(siteId, true);
+        generalLockService.lock(gitLockKey);
+        try {
+            Repository repo = helper.getRepository(siteId, StringUtils.isEmpty(siteId) ? GLOBAL : SANDBOX);
+
+            // Create basic file
+            File file = new File(repo.getDirectory().getParent(), emptyFilePath.toString());
+
+            // Create parent folders
+            File folder = file.getParentFile();
+            if (folder != null && !folder.exists()) {
+                folder.mkdirs();
+            }
+
+            // Create the file
+            if (!file.createNewFile()) {
+                logger.error("Failed to write file to site '{}' path '{}'", siteId, emptyFilePath);
+                throw new ServiceLayerException(format("Error creating new folder at path '{}' site '{}'", newFolderPath, siteId));
+            }
+
+            // Add the file to git
+            if (!helper.addFiles(repo, siteId, emptyFilePath.toString())) {
+                throw new ServiceLayerException(format("Error creating new folder at path '{}' site '{}'", newFolderPath, siteId));
+            }
+
+            String commitId = helper.commitFiles(
+                    repo,
+                    siteId,
+                    helper.getCommitMessage(REPO_CREATE_FOLDER_COMMIT_MESSAGE)
+                            .replaceAll(PATTERN_SITE, siteId)
+                            .replaceAll(PATTERN_PATH, newFolderPath),
+                    helper.getCurrentUserIdent(), emptyFilePath.toString()
+            );
+
+            if (commitId == null) {
+                throw new ServiceLayerException(format("Error creating new folder at path '{}' site '{}'", newFolderPath, siteId));
+            }
+
+            return commitId;
+        } catch (Exception e) {
+            logger.error("Failed to add file to git in site '{}' path '{}'", siteId, newFolderPath, e);
+            throw new ServiceLayerException(format("Error creating new folder at path '{}' site '{}'", newFolderPath, siteId));
```

Comment:
Include the caught exception `e` in the new ServiceLayerException constructor

Reply:
Updated